core/src/autogluon/core/searcher/local_grid_searcher.py

```diff
@@ -0,0 +1,44 @@
+import logging
+
+from sklearn.model_selection import ParameterGrid
+
+from .local_searcher import LocalSearcher
+from ..space import Categorical, Space
+
+__all__ = ['LocalGridSearcher']
+
+logger = logging.getLogger(__name__)
+
+
+class LocalGridSearcher(LocalSearcher):
+    """
+    Grid Searcher that exhaustively tries all possible configurations.
+    This Searcher can only be used for discrete search spaces of type :class:`autogluon.space.Categorical`
```

Comment:
We should not limit grid search to categoricals only. It is possible to sample from a numeric space via arange, [linspace](https://numpy.org/doc/stable/reference/generated/numpy.linspace.html) and [logspace](https://numpy.org/doc/stable/reference/generated/numpy.logspace.html#numpy.logspace). One of the most common searches are actually numeric: learning rates/batch sizes, etc.

Reply:
I'm currently just replicating the logic that was in the original GridSearcher. I agree this could be done, but lets do it in follow-up if it becomes important.